### ISSUE TITLE
align_zero_loss_peak error

### ISSUE THREAD

**Issue by magnunor** [hyperspy/hyperspy#2236]
With the most recent version of HyperSpy (1.5.1) and all the most recent pypi version of the dependencies, the following code gives an error:

```python
import hyperspy.api as hs
s = hs.datasets.artificial_data.get_low_loss_eels_line_scan_signal()
s.align_zero_loss_peak()
```

I get the following error: https://gist.github.com/magnunor/1605e8f4b8979df7819caf633bfc4722

However, the function itself gives the correct result.

This does not happen if I do _not_ grab the most recent version of all the dependencies. So I'm not certain which library is the offending one, but I'm guessing NumPy is a possible one.

Does this error also occur in the most recent bundle and conda-forge version? I'm not able to check on my own computer currently...

**Comment by jat255** [comment]
Testing now with the v1.5.1 bundle

**Comment by ericpre** [comment]
I have just tested it using miniconda on linux, winpython and anaconda on windows, all work fine. Can you try in a conda environment on linux?

**Comment by jat255** [comment]
Sure, updating my conda installation currently

**Comment by ericpre** [comment]
@jat255: I meant @magnunor (both our message appear at the same time!). In any case, another pair of eyes can't hurt!

**Comment by jat255** [comment]
Working for me without error through conda on Linux via a git install, and in a new environment installed with `conda install -c conda-forge hyperspy hyperspyui`

**Comment by jat255** [comment]
Windows bundle works too in my test

**Comment by magnunor** [comment]
I don't currently have time to do extensive testing, but it could be due to the NumPy version:

* Error: 1.17.0 (most recent pypi version)
* Not error: 1.16.2

So if we're unlucky, and the NumPy version on Anaconda is upgraded to 1.17.0, people who use the Anaconda version might get the same error during the short course.

Which NumPy version is in the Windows bundle?

**Comment by magnunor** [comment]
I just tested NumPy 1.16.4 vs. 1.17.0 in the same environment

* 1.17.0 has the error
* 1.16.4 does not have the error

**Comment by jat255** [comment]
My conda installations on Linux have 1.16.4, and the WinPython bundle that I installed has 1.16.4+mkl, so we should be okay during the tutorial unless people try to install with `pip`

**Comment by magnunor** [comment]
Bejoscha in the gitter chat also got the same bug, not certain which operating system he is using. But I'm guessing it affects all systems, as NumPy bugs (in my experience) typically do.

This means that Anaconda installs will (probably) be affected by this whenever the default NumPy changes to 1.17.0, and that pip installs are currently affected by this.

**Comment by ericpre** [comment]
The numpy feedstock has been updated a couple of weeks and the hyperpsy one was fixed in https://github.com/conda-forge/hyperspy-feedstock/pull/39.

**Comment by norskie** [comment]
I have the same problem using Ubuntu with a fresh install of miniconda, reverting to v1.41 of hyperspy seems to fix the problem for me. Many different hyperspy functions does not work, and the latest traceback seems to always point to Numpy. 

One may i.e reproduce the problem by running the following notebook: 
https://github.com/pyxem/pyxem-demos/blob/master/01%20GaAs%20Nanowire%20-%20Data%20Inspection%20-%20Preprocessing%20-%20Unsupervised%20Machine%20Learning.ipynb

And the traceback.
[https://imgur.com/a/2cwwyoT](url)

**Comment by ericpre** [comment]
The issue with the recent numpy release 1.17.* is the activation of the new `__array_function__` protocol, see https://docs.scipy.org/doc/numpy/release.html#changes.

In the mean time, this can be deactivated by setting an environment variable (https://numpy.org/neps/nep-0018-array-function-protocol.html#implementation):
> In NumPy 1.17, the protocol will be enabled by default, but can be disabled with NUMPY_EXPERIMENTAL_ARRAY_FUNCTION=0.

Installing numpy 1.16 will work too since in this version the `__array_function__` protocol is disable.

@norskie: this is surprising that downgrading to 1.4.1 works as workaround too. Maybe, by doing this, it also downgraded numpy but I don't see why and I can't reproduce.

**Comment by francisco-dlp** [comment]
Fixed in #2247
